book/6-modelling/cv-metrics.md

```diff
@@ -0,0 +1,80 @@
+
+FPS should be calculated based off either model's or the software architecture's inference speed. This is obvious but often overlooked. When delivering the model, the onus is on the AI Engineer to communicate how the FPS was measured to manage expectations.
+
+It is also important to consider what affects FPS. The first factor would be model size. A model that is deep, and hence tend to be larger in size, requires a longer processing time. Another factor is image size. Larger image size will take longer for inference. For example, a 416x416 image will be faster than 1080x960 image. While again this is obvious, it is also often overlooked by AI Engineers when reading papers that purport higher inference speeds, but downplays the role that smaller images have on inference speed.  
+
+## Adjusting Thresholds for Inference - Confidence and Non-max Suppression (NMS)
+
+When a model makes an inference, it will give a confidence level of the object's class. If the confidence does not meet or exceed the threshold, then it will not be included. 
+
+Similarly, when a model makes duplicative inference on the same object, meaning it has two or more bounding boxes on the same object, there has to be a mechanism to suppress these duplicative boxes.
+
+Thus, the engineer may use the Confidence and/or NMS threshold to ensure to adjust the precision and recall of the model. 
+
+To show this, consider the output below. Such a table can be generated by running the model on a combination of possible threshold values. 
+
+| Confidence 	| NMS  	| IOU  	| Precision 	| Recall 	| F1 Score 	|
+|------------	|------	|------	|-----------	|--------	|----------	|
+| 0.3        	| 0.1  	| 0.5  	| 0.697     	| 0.752  	| 0.724    	|
+| 0.5        	| 0.1  	| 0.5  	| 0.750     	| 0.732  	| 0.741    	|
+| 0.7        	| 0.1  	| 0.5  	| 0.811     	| 0.697  	| 0.750    	|
+| 0.3        	| 0.4  	| 0.5  	| 0.640     	| 0.789  	| 0.707    	|
+| 0.5        	| 0.4  	| 0.5  	| 0.718     	| 0.759  	| 0.738    	|
+| 0.7        	| 0.4  	| 0.5  	| 0.794     	| 0.717  	| 0.754    	|
+| 0.3        	| 0.7  	| 0.5  	| 0.579     	| 0.800  	| 0.672    	|
+| 0.5        	| 0.7  	| 0.5  	| 0.682     	| 0.768  	| 0.722    	|
+| 0.7        	| 0.7  	| 0.5  	| 0.774     	| 0.723  	| 0.747    	|
+
+It is clear here that Confidence and NMS threshold can appear to be at odds with each other. As Confidence Threshold is increases, the Precision increases, but Recall is lower. Conversely, a higher NMS threshold gives better recall.
+
+## Prioritising Precision or Recall
+
+As stated above, the AI Engineer has to continually engage with the product owner in determining which to prioritise. Obviously, they can also rely on F1-Score to harmonise the two. However, do keep in mind the following:
+
+1. Choose Precision if the priority of the model is to identify objects. 
+2. Choose Recall if the priority of the model is identify objects and their class correctly.
+
```

Comment:
I'm a bit unclear as to the difference between 1 and 2 as stated here, could you elaborate?